### PR TITLE
414 and 418 my collaborations and a common json collaborations format

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Collaboration details as requested in DEX 418 and keeping the same for create response/ collaborations GET/ my collaborations GET    
- Removed conecpt of an overall view state for the collaboration, tere is none, it is different for each user
- Removed code no longer used
---

**REST API Updates **

common Collaborations output format:

{'members':
    {
    'e3608816-7b3f-4062-9180-5bf0f6b7138f': 
        {
             'full_name': 'First Middle Last', 
             'email': 'researcher1@localhost', 
             'guid': 'e3608816-7b3f-4062-9180-5bf0f6b7138f', 
            'viewState': 'approved', 
            'initiator': True
        }, 
    '122a8a56-ccc3-45a7-a8f8-4adfbaa9f581': 
        {
            'full_name': 'First Middle Last', 
            'email': 'researcher2@localhost', 
            'guid': '122a8a56-ccc3-45a7-a8f8-4adfbaa9f581', 
           'viewState': 'pending', 
           'initiator': False
       }
   }, 
   'updated': '2021-08-11T09:25:11.790075+00:00', 
   'created': '2021-08-11T09:25:11.782742+00:00', 
   'guid': 'f95ccdb1-625a-45fa-8638-65df31fc30c7'
}

```
@colinwkingen  to review the code, @brmscheiner  to review the json format

### CODE REVIEW

File: app/modules/collaborations/resources.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ class Collaborations(Resource):
         },
     )
     @api.parameters(PaginationParameters())
-    @api.response(schemas.BaseCollaborationSchema(many=True))
+    @api.response(schemas.DetailedCollaborationSchema(many=True))
```
Comment (hwindsor) [inline app/modules/collaborations/resources.py:50]: This was the one line needed to make 418 the same so it really didn't need a separate PR

File: tests/modules/collaborations/resources/test_collaboration_create.py [hunk 2]
```diff
@@ -16,31 +25,44 @@ def test_create_collaboration(
...
     collab_utils.create_collaboration(flask_app_client, researcher_1, data, 400, resp_msg)
 
-    from app.modules.collaborations.models import Collaboration
-
     collab = None
     try:
         data = {'user_guid': str(researcher_2.guid)}
         resp = collab_utils.create_collaboration(flask_app_client, researcher_1, data)
-        collabs = Collaboration.query.all()
-        collab = collabs[0]
```
Comment (hwindsor) [inline tests/modules/collaborations/resources/test_collaboration_create.py:26]: from your comment @colinwkingen on the previous PR

File: app/modules/users/schemas.py [hunk 3]
```diff
@@ -94,10 +94,13 @@ class PersonalUserSchema(DetailedUserSchema):
     that can be edited by the currently logged in user.
     """
 
+    collaborations = base_fields.Function(User.get_collaborations_as_json)
+
     class Meta(DetailedUserSchema.Meta):
         fields = DetailedUserSchema.Meta.fields + (
             User.unprocessed_asset_groups.__name__,
             User.unprocessed_sightings.__name__,
+            'collaborations',
```
Comment (hwindsor) [inline app/modules/users/schemas.py:103]: Have a think about this, do we really want the full details for all collaborations each time the user conects?

Comment (colinwkingen) [reply app/modules/users/schemas.py:103]: I think it's okay and can be changed later if it becomes too expensive. I imagine on most platforms users will only have a handful of collaborations.

**Review by colinwkingen — APPROVED** [review]
Looks great. Tightens up the tests quite a bit.

**Pull request merged**